### PR TITLE
Bump Apollo Codgen to version 0.10.8. Simplify compiler test fixture …

### PR DESCRIPTION
**Pull request by BenSchwab**
…generation.

Apollo-codegen 0.10.8 supports field arguments, which are needed to generate
cache keys.

The process of updating the code-gen version was very manual as the
apollo code-gen tests run from pure IR (TestQuery.json) files.

When we bump the codegen, we will want to regenerate the TestQuery.json
files. This diff extracts the TestQuery.graphql for each of the compiler
tests and creates a simple gradle plugin to execute apollo-codegen
with the *Local* version of apollo-codegen.

While it would be nice to have this happen automatically, doing so
would require following a approach similar to the apollo-gradle-plugin
which installs apollo-codegen to a local repository. Sharing this code
would require a more significant refactor to avoid logic duplication.

Instead, whenever we update the codegen plugin, we can run
`./gradlew :apollo-compiler:updateTestIRFiles`.
This will record the local apollo codegen version used to `apollo-codegen.properties`
to force review acknowledgment of the version used.

This diff mostly unifies the compiler tests on one
schema (https://github.com/apollographql/starwars-server).  Custom logic needed for certain tests was injected into the schema. 

In cases where the test needs something completely custom, you can place a schema.json file in the test folder.

Closes [225](https://github.com/apollographql/apollo-android/issues/225)

### CODE REVIEW

File: apollo-compiler/src/test/graphql/com/example/directives/TestQuery.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "operations": [
     {
+      "filePath": "/Users/ben_schwab/airapps/apollo-android/apollo-compiler/src/test/graphql/com/example/directives/TestQuery.graphql",
```
Comment (brianPlummer) [inline apollo-compiler/src/test/graphql/com/example/directives/TestQuery.json]: "/Users/ben_schwab/" is this intentional?

Comment (BenSchwab) [inline apollo-compiler/src/test/graphql/com/example/directives/TestQuery.json]: It seems that the apollo-codgen records this as an absolute path now. I could go and remove all these lines, but they don't seem to have an impact on the codegen.

Comment (brianPlummer) [inline apollo-compiler/src/test/graphql/com/example/directives/TestQuery.json]: no worries then.  just double-checking :)

File: apollo-compiler/gradle/update-test-IR-files.gradle [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    def dir = new File("apollo-compiler/src/test/graphql/com/example")
+    dir.eachFile(FileType.DIRECTORIES) {
+      testDir -> generateIRFile(testDir)
+    }
+    logApolloVersion(apolloVersion)
+  }
+
+  private static def logApolloVersion(apolloVersion) {
+    File outputFile = new File('apollo-compiler/src/test/graphql/apollo-codegen.properties')
```
Comment (marwanad) [inline apollo-compiler/gradle/update-test-IR-files.gradle]: does this file get used after?

File: apollo-gradle-plugin/src/main/java/com/apollographql/android/gradle/ApolloCodeGenInstallTask.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
 public class ApolloCodeGenInstallTask extends NpmTask {
   static final String NAME = "installApolloCodegen";
   private static final String INSTALL_DIR = "node_modules/apollo-codegen";
-  private static final String APOLLOCODEGEN_VERSION = "0.10.5";
+  private static final String APOLLOCODEGEN_VERSION = "0.10.8";
```
Comment (marwanad) [inline apollo-gradle-plugin/src/main/java/com/apollographql/android/gradle/ApolloCodeGenInstallTask.java]: Would it make more sense to add it to the companion object in `GraphQLCompiler` and use that instead?

Comment (BenSchwab) [inline apollo-gradle-plugin/src/main/java/com/apollographql/android/gradle/ApolloCodeGenInstallTask.java]: Yeah, good call. Seems like a better fit as this is something the Compiler is "aware" of, albeit indirectly

**Pull request merged**